Objects/typeobject.c

```diff
@@ -3066,6 +3081,67 @@ PyType_GetSlot(PyTypeObject *type, int slot)
     return  *(void**)(((char*)type) + slotoffsets[slot]);
 }
 
+PyObject *
+PyType_GetModule(PyTypeObject *type) {
+    PyHeapTypeObject* et = (PyHeapTypeObject*)type;
+    if (!et->ht_module) {
+        PyErr_BadInternalCall();
+        return NULL;
+    }
+    return et->ht_module;
+
+}
+
+PyTypeObject *
+PyType_DefiningTypeFromSlotFunc(PyTypeObject *type, int slot, void *func) {
+    Py_ssize_t pos = 0;
+    PyObject *mro = type->tp_mro;
+
+    assert(PyType_HasFeature(type, Py_TPFLAGS_HEAPTYPE));
+
+    while(1) {
+        if (PyType_GetSlot(type, slot) == func) {
+            break;
+        }
+
+        type = (PyTypeObject *)PyTuple_GetItem(mro, pos);
```

Comment:
What is the guarantee that prevents the two tuple item accesses (here and below) from running out of bounds?
Could that be worth a comment, at least?

Reply:
A `for` loop might be better: first use `PyTuple_Size` to also check the type, then use the faster `PyTuple_GET_ITEM`.